providers/openshift/provider.go

```diff
@@ -586,3 +580,18 @@ func request(client *http.Client, req *http.Request) (*simplejson.Json, error) {
 	}
 	return data, nil
 }
+
+func getKubeAPIURL() *url.URL {
+	host := "https://" + os.Getenv("KUBERNETES_SERVICE_HOST")
+	if len(host) == 0 {
+		return &url.URL{
+			Scheme: "https://",
+			Host:   "kubernetes.default.svc",
+		}
+	}
+	url, err := url.Parse(host)
+	if err != nil { // this should never happen
+		log.Printf("ERROR: failed to retrieve Kubernetes API URL: %s", err)
```

Comment:
`panic` or fallback to a safe `*url.URL`.

Reply:
I worked around that whole url parsing thing by simply building it from scratch instead.